src/kontrol/__main__.py

```diff
@@ -78,6 +73,7 @@ def main() -> None:
     sys.setrecursionlimit(15000000)
     parser = _create_argument_parser()
     args = parser.parse_args()
+    args = read_toml_args(parser, args, sys.argv[2:]) if hasattr(args, 'config_file') else args
```

Comment:
What's the reason behind using `sys.argv` directly?

Reply:
In this setup, I think it no longer makes sense to let `argparse` handle default values, i.e. all `default` arguments should be left unspecified. If you can in addition also ensure that `None` is not a valid value for any option (which I think should be straightforward), then you can do the following:

1. First, parse the command line options to get the config file location. Keep the `Namespace` object around for later.
2. Handler 1: Defaults. Create a `dict` with default values. Pass on the result to the next handler.
3. Handler 2: Config file. Parse and process the config, update the `dict`. Pass on the result to the next handler.
4. Handler 3: Command line arguments. Update the `dict` from the `Namespace`, skipping keys with value `None`.

Here, each handler takes a `dict` and updates it with values obtained from a given source. (Note that the order can also be reversed if you'd like, i.e. start with the source with the highest priority.)

Reply:
@tothtamas28 @ovatman do you think we can get this PR merged with some minimal changes, before @nwatson22  merges the `pyk` default value update (https://github.com/runtimeverification/pyk/pull/916) to make these changes easier to merge?